### PR TITLE
Feat/truncate

### PR DESCRIPTION
**Pull request by ewendai**
### Done checklist
- [x] docs/SPEC.md updated
- [x] Test cases written

#1533

### CODE REVIEW

File: docs/SPEC.md [hunk 1]
```diff
@@ -1132,6 +1132,12 @@ These are builtin functions that all take a single `time` argument and return an
 * `month` int
     Month returns the month of the year for the provided time in the range `[1-12]`.
 
+#### truncate
+
+`date.truncate` takes in a time t and a Duration unit and returns the given time 
+truncated to the given unit.
+
+Example: `truncate(t: "2019-06-03T13:59:01.000000000Z", unit: 1s)` returns time `2019-06-03T13:59:01.000000000Z`
```
Comment (affo) [inline docs/SPEC.md]: Could you give a more insightful example? For  example with minute or hour?

If you add examples as in your unit tests, that would be great!

File: stdlib/universe/universe.flux [hunk 2]
```diff
@@ -81,7 +82,7 @@ builtin inf
 builtin length // length function for arrays
 builtin linearBins
 builtin logarithmicBins
-builtin sleep // sleep is the identity function with the side effect of delaying execution by a specified duration.
+builtin sleep // sleep is the identity function with the side effect of delaying execution by a specified duration
```
Comment (affo) [inline stdlib/universe/universe.flux]: Why is this here? A bad rebase or something else?

File: docs/SPEC.md [hunk 3]
```diff
@@ -3567,6 +3573,18 @@ r0 = from(bucket:"telegraf/autogen")
...
+
+Truncates all entries in a given time column to a specified unit.
+
+Example: 
+```
+from(bucket:"telegraph/autogen")
+    |> truncateTimeColumn(timeColumn: "_time", unit: 1s)
+```
+
+Currently, `truncateTimeColumn` only works properly if the given `timeColumn` is "_time".
```
Comment (affo) [inline docs/SPEC.md]: Yeah, I see your point. I would rephrase into:

```
Currently, `truncateTimeColumn` only works with the default time column `_time`.
```

and I would change the example to:

```
from(bucket:"telegraph/autogen")
    |> truncateTimeColumn(unit: 1s)
```

If we omit `timeColumn` from the doc, it is not a big deal, because, for now, nobody can use it.

You should also add a table with the parameters as the other functions, don't you think?

As a final note, I actually see a way to make it work with any column:

```
truncateTimeColumn = (timeColumn="_time", unit, tables=<-) => tables
        |> map(fn:(r) => ({r with __truncated_time: date.truncate(t: r._time, unit: unit)}))
        |> duplicate(column: "__truncated_time", as: timeColumn) // overwrites timeColumn
        |> drop(columns: ["__truncated_time"])
```

If the  user already has a column called `__truncated_time` that could be an issue though 😓 . What do you think?

FYI @aanthony1243 .

File: stdlib/date/date.go [hunk 4]
```diff
@@ -147,6 +146,29 @@ func init() {
...
+			"truncate",
+			semantic.NewFunctionPolyType(semantic.FunctionPolySignature{
+				Parameters: map[string]semantic.PolyType{"t": semantic.Time, "unit": semantic.Duration},
+				Required:   semantic.LabelSet{"t", "unit"},
+				Return:     semantic.Time,
+			}),
+			func(args values.Object) (values.Value, error) {
+				v, ok := args.Get("t")
+				u, unitOk := args.Get("unit")
+				if !ok {
```
Comment (affo) [inline stdlib/date/date.go]: Can you move this directly below line 157 and directly use `GetRequiredTime`/`Duration`. That would simplify logic.

File: stdlib/universe/universe.flux [hunk 5]
```diff
@@ -1,6 +1,7 @@
 package universe
 
 import "system"
+import "date"
```
Comment (affo) [inline stdlib/universe/universe.flux]: this is great. Makes me think why don't we `import "math"` but we add it programmatically to the prelude @aanthony1243 .

File: stdlib/universe/universe.flux (hunk 2 above)
Comment (ewendai) [reply stdlib/universe/universe.flux]: I deleted the period

File: stdlib/date/date.go (hunk 4 above)
Comment (ewendai) [reply stdlib/date/date.go]: `args.GetRequiredTime` and `args.GetDuration` don't pop up when I type it in?

Comment (affo) [reply stdlib/date/date.go]: It's here: https://github.com/influxdata/flux/blob/c7e454395264dcf1cb23f9889ef7bbbaa947fc4c/compile.go#L682

File: docs/SPEC.md [hunk 6]
```diff
@@ -1137,7 +1137,10 @@ These are builtin functions that all take a single `time` argument and return an
 `date.truncate` takes in a time t and a Duration unit and returns the given time 
 truncated to the given unit.
 
-Example: `truncate(t: "2019-06-03T13:59:01.000000000Z", unit: 1s)` returns time `2019-06-03T13:59:01.000000000Z`
+Examples: 
+- `truncate(t: "2019-06-03T13:59:01.000000000Z", unit: 1s)` returns time `2019-06-03T13:59:01.000000000Z`
+- `truncate(t: "2019-06-03T13:59:01.000000000Z", unit: 1m)` returns time `2019-06-03T13:59:00.000000000Z`
+- `truncate(t: "2019-06-03T13:59:01.000000000Z", unit: 1h)` returns time `2019-06-03T13:00:00.000000000Z`
```
Comment (affo) [inline docs/SPEC.md]: Cool, thank you :)

File: stdlib/universe/universe.flux (hunk 5 above)
Comment (aanthony1243) [reply stdlib/universe/universe.flux]: never thought about that.  I'm going to test it

File: docs/SPEC.md (hunk 3 above)
Comment (ewendai) [reply docs/SPEC.md]: `date.truncate` still takes in only `_time`

this just changes `timeColumn` from the column you want to truncate to what the resulting column is named

File: stdlib/date/date.go (hunk 4 above)
Comment (ewendai) [reply stdlib/date/date.go]: I think `args` is `*TableObject` not `Arguments`, so how would that work?

Comment (affo) [reply stdlib/date/date.go]: Oh I see, sorry about that! This code is ok though :)

File: stdlib/universe/universe.flux (hunk 2 above)
Comment (affo) [reply stdlib/universe/universe.flux]: This is  mistery. In `master` is there. I  think that if  you rebase that would go away. Anyways this shouldn't be a problem.

Comment (affo) [reply stdlib/universe/universe.flux]: Oooooh now I got it. The period `.` 😅 

I haven't noticed that. I am using the split view in Github and it's harder to catch that.

File: docs/SPEC.md (hunk 3 above)
Comment (affo) [reply docs/SPEC.md]: Yeah, I am sorry, I meant:

```
truncateTimeColumn = (timeColumn="_time", unit, tables=<-) => tables
        |> duplicate(column: timeColumn, as: "__truncated_time")
        |> map(fn:(r) => ({r with __truncated_time: date.truncate(t: r.__truncated_time, unit: unit)}))
        |> duplicate(column: "__truncated_time", as: timeColumn) // overwrites timeColumn
        |> drop(columns: ["__truncated_time"])
```

Anyways, this is a workaround that could overwrite a column with the same name by the user. We need to support labels for this #1122 .

Only brainstorming 😄
